src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultViewGenerator.java

```diff
@@ -70,16 +75,86 @@ private boolean itemsRequireFormatting(List<SearchResultViewHeaderGroup> headerG
+    private CaseDetails filterResultsByAuthorisationAndUserRole(String useCase, CaseDetails caseDetails, String caseTypeId, List<String> requestedFields) {
+        Set<String> roles = userRepository.getUserRoles();
+        CaseTypeDefinition caseTypeDefinition = getCaseTypeDefinition(caseTypeId);
+        SearchResult searchResultDefinition = searchResultDefinitionService.getSearchResultDefinition(caseTypeDefinition, useCase, requestedFields);
+        HashMap<String, String> searchFields = getSearchResultDefinitionFieldUserRoleAndField(searchResultDefinition);
+
+        Iterator<String> caseFieldsUseCase = caseDetails.getData().keySet().iterator();
+        while (caseFieldsUseCase.hasNext()) {
+            Object caseField = caseFieldsUseCase.next();
+            if (useCase != null) {
+                if (!searchFields.containsKey(caseField)) {
+                    caseFieldsUseCase.remove();
+                    caseField = null;
+                }
+                String role = searchFields.get(caseField);
+                if (role != null && !roles.contains(role)) {
+                    caseFieldsUseCase.remove();
+                    caseField = null;
+
+                }
+            }
+            if (caseField != null) {
+                Optional<CaseFieldDefinition> caseFieldDefinition = caseTypeDefinition.getCaseField(caseField.toString());
+                if (caseFieldDefinition.isPresent() && caseFieldDefinition.get().isComplexFieldType()) {
+                    if (!filterResultsByAuthorisationAccess(caseFieldDefinition)) {
+                        caseFieldsUseCase.remove();
+                    }
+                } else {
+                    if (!accessControlService.canAccessCaseFieldsWithCriteria(
+                        JacksonUtils.convertValueJsonNode(caseField),
+                        caseTypeDefinition.getCaseFieldDefinitions(),
+                        roles,
+                        CAN_READ)) {
+                        caseFieldsUseCase.remove();
+                    }
+                }
+
+            }
+
+        }
+        return caseDetails;
+    }
+
+    private Boolean filterResultsByAuthorisationAccess(Optional<CaseFieldDefinition> caseFieldDefinition) {
+        if (caseFieldDefinition.isPresent()) {
+            List<AccessControlList> accessControlLists = caseFieldDefinition.get().getAccessControlLists();
+            Set<String> roles = userRepository.getUserRoles();
+            if (!caseFieldDefinition.get().isMetadata()) {
+                for (AccessControlList accessControlList : accessControlLists) {
+                    for (String role : roles) {
+                        if (accessControlList.getRole().equals(role)) {
+                            return accessControlList.isRead();
+                        }
+                    }
+                }
+                return false;
+            }
+            return true;
+        }
+        return false;
```

Comment:
Requires testing but I think can be simplified to something like this?
```suggestion
        if (!caseFieldDefinition.isMetadata()) {
            return userRepository.getUserRoles().stream()
                .anyMatch(role -> caseFieldDefinition.getAccessControlListByRole(role).map(AccessControlList::isRead).orElse(false));
        }
        return true;
```